### DISCUSSION TITLE
제출 관련 이슈 Discussion

### DISCUSSION THREAD

**Discussion by Kyushik** [reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity#23] [General]
제출 관련하여 Issue가 있는 경우 해당 Discussion에 댓글을 남겨주세요! 
최대한 빠르게 대응하도록 하겠습니다!

**Comment by paekgga** [comment]
안녕하세요. Tensorflow 모델로 제출하려고 하는데 No module named tensorflow 에러가 발생하네요.. 이경우 어떻게 해야되나요?

**Comment by Kyushik** [comment]
@paekgga 
안녕하세요! 안그래도 tensorflow로 제출하는 경우 이슈가 있는 것으로 확인하였습니다. TF의 경우 1버전으로 제출하시는 분도 있으시고 2버전으로 제출하시는 분도 있으셔서 처리가 조금 어려운 상황인데요! 평가 담당해주시는 AIFactory에서 다음과 같이 내용 전달주셨습니다!

python api로 제출할 때 library 디펜던시의 경우 참가자분들이 
`pip install -r requirements.txt --target=./lib`
위 명령어로 원하는 라이브러리를 설치하고 zip 파일에 같이 담아서 제출하면 추가하여 사용할 수 있습니다. 

위 내용대로 진행해서 제출 부탁드립니다! 감사합니다! 
이상있는 경우에는 다시 댓글 남겨주시면 확인해보고 빠르게 처리할 수 있도록 하겠습니다.

**Reply by paekgga** [reply]
답변 감사합니다 :)

/lib 폴더에 라이브러리를 직접 설치하고 압축해서 제출해야하는건가요?
아니면 requirements.txt 파일에만 추가 dependency 리스팅해서 제출하면되나요?

직접 /lib폴더에 제출하는경우 패키지만해도 200MB초과로 제출이 안됩니다~

**Comment by Kyushik** [comment]
혹시 Tensorflow 2 사용하시고 계실까요? 지금 TF가 TF1으로 제출하시는 분이랑 2로 제출하시는 분이 계셔서 이슈가 있는 상황입니다! ㅜㅜㅜ 현재 ./lib가 없으면 tensorflow 2.7로 설치된 라이브러리 참조하도록 수정했습니다!

**Reply by paekgga** [reply]
Tensorflow ver 1을 사용하고 있는데, Tensorflow 2 에서 실행될수있도록 코드 수정해서 제출해보도록 하겠습니다. 추가로, 베이스라인 코드에 있는 skimage (scikit-image) 패키지도 셋팅 가능할까요?

**Reply by Kyushik** [reply]
skimage의 경우 사용하시는 분들이 많으실 수 있어서 한번 설치 가능한지 말씀드려보겠습니다!

**Reply by Kyushik** [reply]
scikit-image==0.18.3
위의 버전으로 세팅 진행했습니다!

**Reply by paekgga** [reply]
감사합니다. scikit-image 설치확인하였습니다. Tensorflow 2.7 라이브러리 관련해서 CuDNN 버전 미스매칭 문제가 있는것 같습니다. 에러내용은 아래와같습니다. 확인해주실수 있을까요?

**Reply by paekgga** [reply]
E tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded runtime CuDNN library: 8.0.5 but source was compiled with: 8.1.0. CuDNN library needs to have matching major version and equal or higher minor version. If using a binary install, upgrade your CuDNN library. If building from sources, make sure the library loaded at runtime is compatible with the version specified during compile configuration. 2021-11-17 12:59:51.605200: W tensorflow/core/framework/op_kernel.cc:1745] OP_REQUIRES failed at conv_ops_fused_impl.h:560 : UNKNOWN: Failed to get convolution algorithm. This is probably because cuDNN failed to initialize, so try looking to see if a warning log message was printed above. 2021-11-17 12:59:51.607423

**Reply by Kyushik** [reply]
네네 말씀하신대로 CuDNN 이슈인 듯 합니다! 해당 내용 AIFactory 측에 문의했습니다!! ;)

**Reply by Kyushik** [reply]
방금 AI Factory 측에서 연락 받았습니다! 평가를 위한 Tensorflow 버전을 2.4로 변경했더니 제출하신 코드 잘 동작한다고 합니다! 한번 제출해보시면 좋을 것 같습니다! :)

**Reply by paekgga** [reply]
감사합니다. 정상 제출되는 것 확인하였습니다.
그런데, 점수결과가 저희 자체 테스트결과와 너무 상이하게 나오네요.
자체테스트는 제출한 코드를 Baseline/code/Inference.py 파일로 여러번 반복해서 테스트를 진행하였습니다.
라이브러리 버전은 위 말씀주신 버전에 모두 맞추어 진행하였습니다.
채점 시스템이 동일한 방식으로 작동할 것 같은데, 어떤 이슈사항이 있는지 파악하기가 어렵네요..

**Reply by Kyushik** [reply]
일단 제가 테스트해봤을 때 테스트환경에 문제는 없는 듯 합니다! 
몇가지 의심해볼 수 있는 요소들이 있을 것 같은데요! 
우선 상태 전처리를 학습 및 테스트 할 때랑 동일하게 하셨는지 봐주시면 좋을 듯 하구요..!
다음으로는 학습 환경이 혹시 서버일까요..? --no graphics 옵션을 사용하셨거나 모니터가 없는 서버에서 학습을 수행하는 경우 이미지 입력이 제대로 받아지지 않는 경우가 있습니다. 이렇게 학습했다가 제대로 이미지 입력이 들어오는 테스트 환경에서 검증해서 성능이 제대로 나오지 않을 수 있다.. 라고 의심해볼 수 있을 것 같습니다. 
위에 두가지 사항에 대해서 한번 확인 부탁드리겠습니다!

**Reply by paekgga** [reply]
개인적인 이슈사항인 것 같아 메일로 답변 및 추가문의 드렸습니다. 확인부탁드립니다 :)

**Comment by LUKE-hyungjin** [comment]
안녕하세요. JORLDY로 학습 후 제출을 했는데 아래의 사진과 같은 에러가 나옵니다. Agent.py의 load_model에 weight_file_path과 map_location 이 부분이 틀린 것 같은데, 페이스북에 올려주신 노션에 "입력으로 들어오는 weight_file_path를 그대로 사용하셔야 합니다"의 의미를 잘 모르겠습니다. 어떤 경로로 작성해야 하는 것인지 알려주실 수 있나요? 
![image](https://user-images.githubusercontent.com/68213792/143536641-264faf49-6cce-4809-9edd-5e32ac3c1406.png)

**Reply by Kyushik** [reply]
![image](https://user-images.githubusercontent.com/11716370/143537615-284531bb-57bf-4d6e-acfc-9d2cad7f20a1.png)

위의 그림에서 나오듯이 weight_file_path를 torch.load의 입력 인자로 그대로 사용해주시면 됩니다! 왠지 에러 메시지 보면 파일 로드 자체는 시도했는데 그 안에 있는 키가 틀린 느낌이네요..? 혹시 학습을 수행한 모델의 네트워크와 Agent.py에 포함된 네트워크에 차이가 없는지 한번 확인해주시면 좋을 것 같습니다!

**Reply by LUKE-hyungjin** [reply]
답변 감사합니다. 실제로 학습을 수행한 모델의 네트워크와 Agent.py의 네트워크가 달라 문제가 발생했습니다.  
혹시 jorldy 사용하시는 분들은 Agent.py에 ContinuousPolicyValue쪽을 보시고 학습한 네트워크를 수정, 추가하시면 될 것 같습니다!

**Comment by chaningdev** [comment]
![image](https://user-images.githubusercontent.com/84086347/143665916-608bac3f-da38-4def-b2e7-1b9f81101fee.png)


11시 48분경에 3번 제출을 진행하였으나, 사진과 같이 첫 두번은 잘 되었지만 마지막 채점의 시간대가 조금 이상한 것 같습니다. 
확인 부탁드립니다.  세번 모두 성공이라고 뜬 후에 진행한 채점입니다.

고생하십니다!

**Comment by paekgga** [comment]
현재 채점시스템이 작동하지 않는것 같습니다. 확인부탁드립니다~

**Reply by Kyushik** [reply]
헛 바로 문의해보겠습니다! 혹시 어떤식으로 작동하지 않는지 말씀해주시면 좀더 참고가 될 것 같습니다!

**Reply by paekgga** [reply]
감사합니다 :) 제출한 내역이 장시간 채점중상태에서 멈춰있는 상태입니다

**Reply by paekgga** [reply]
해결확인했습니다! 감사합니다

**Reply by Kyushik** [reply]
아?? 이제 잘 되나요? 넵! 감사합니다!

**Reply by paekgga** [reply]
네네 잘되고있습니다. 감사합니다
